From a555204c3617438250cded94d084b4dec1aaf551 Mon Sep 17 00:00:00 2001
From: anthraxx <levente@leventepolyak.net>
Date: Wed, 23 May 2018 00:26:35 +0200
Subject: [PATCH] libc: respect zip/deps.mk for zlib/libzip

Additionally respect standard caller {C,CPP,LD}FLAGS plus the
deps.mk relaed LINK
---
 libr/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libr/Makefile b/libr/Makefile
index 0b39b48c02..0d02577081 100644
--- a/libr/Makefile
+++ b/libr/Makefile
@@ -3,6 +3,7 @@ include config.mk
 include ../config-user.mk
 include ../mk/platform.mk
 include ../mk/${COMPILER}.mk
+include ${SHLR}/zip/deps.mk
 
 DESTDIR:=$(DESTDIR)
 PREFIX?=${PWD}/../prefix
@@ -91,10 +92,9 @@ E+=../shlr/capstone/libcapstone.${EXT_AR}
 endif
 
 libr.${EXT_SO}: .libr
-	$(CC) -fvisibility=hidden $(MLFLAGS) -shared -dynamiclib -o libr.${EXT_SO}  \
+	$(CC) -fvisibility=hidden $(MLFLAGS) ${LINK} ${CFLAGS} ${CPPFLAGS} ${LDFLAGS} -shared -dynamiclib -o libr.${EXT_SO}  \
 		.libr/*.o \
 		../shlr/gdb/lib/libgdbr.${EXT_AR} ../shlr/java/libr_java.${EXT_AR} \
-		../shlr/zip/librz.${EXT_AR} \
 		../shlr/libr_shlr.${EXT_AR} $(E)
 
 else
